packages/ragbits-core/src/ragbits/core/audit/cli_span.py

```diff
@@ -0,0 +1,78 @@
+import time
+from typing import Optional
+
+from rich.tree import Tree
+
+
+class CLISpan:
+    """
+    CLI Span represents a single operation within a trace.
+    """
+
+    def __init__(self, name: str, parent: Optional["CLISpan"] = None):
+        """
+        Constructs a new CLI Span.
+        Sets the start time of the span - the wall time at which the operation started.
+        Sets the span status to 'started'.
+
+        Args:
+            name: The name of the span.
+            parent: the parent of initiated span.
+        """
+        self.name = name
+        self.parent = parent
+        self.start_time: float = time.time()
+        self.end_time: float | None = None
+        self.children: list[CLISpan] = []
+        self.status: str = "started"
+
+    def end(self) -> None:
+        """Sets the current time as the span's end time.
+        The span's end time is the wall time at which the operation finished.
+        Only the first call to `end` should modify the span,
+        further calls are ignored.
+        """
+        if self.end_time is None:
+            self.end_time = time.time()
+
+    def to_dict(self) -> dict:
```

Comment:
is it used somewhere?

Reply:
no. i needed it for checks. now deleted.